bentoml/_internal/bento/docker.py

```diff
@@ -68,78 +70,78 @@ class ImageProvider(object):
     Example::
 
     from bento import svc
-    from bentoml.utils import builtin_docker_image
 
     svc.build(
         ...
         env=dict(
-            docker_options={
-                'base_image': builtin_docker_image('debian', '3.8', gpu=True)
-            }
+            docker=dict(
+                distro='debian',
+                python='3.8',
+                gpu=True
+            )
         )
     )
     """
 
     def __init__(
         self,
-        distro: str,
+        distro: t.Optional[
+            Literal["slim", "ami2", "alpine", "centos7", "centos8"]
+        ] = None,
         python_version: t.Optional[str] = None,
-        gpu: bool = False,
+        gpu: t.Optional[bool] = None,
```

Comment:
this can just be 
```python
gpu: bool = False
```
If users want to use GPU they can then specify `gpu=True`

Reply:
Actually, this is because we want to allow upstream to call it with `gpu=None`.

Let's say the user has the following code:
```
svc.set_build_options( 
  env = dict(
    docker = dict(
        base_image = ...
    )
  )
)
```

We want to distinguish if the user has provided other options like `distro="slim"` at the same time, and provide the user a proper warning. But if we have `gpu=False, distro='slim'` as the default, we won't be able to tell if it's user-specified or if it's just the default.

Reply:
noted